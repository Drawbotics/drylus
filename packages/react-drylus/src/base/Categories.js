const Categories = {
  BRAND: 'BRAND',
  SUCCESS: 'SUCCESS',
  INFO: 'INFO',
  WARNING: 'WARNING',
  DANGER: 'DANGER',
};


export default Categories;
