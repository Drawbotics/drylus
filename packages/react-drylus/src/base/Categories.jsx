const Categories = {
  PRIMARY: 'PRIMARY',
  SUCCESS: 'SUCCESS',
  INFO: 'INFO',
  WARNING: 'WARNING',
  DANGER: 'DANGER',
};


export default Categories;
