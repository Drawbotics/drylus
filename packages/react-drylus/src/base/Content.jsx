import React from 'react';
import { css, cx } from 'emotion';
import PropTypes from 'prop-types';
import sv from '@drawbotics/style-vars';


const styles = {
  root: css`
    background: ${sv.backgroundColor};
  `,
  fullHeight: css`
    flex: 1;
  `,
  children: css`
    max-width: ${sv.maxWidthLarge};
    margin: auto;
  `,
  fullWidth: css`
    max-width: none;
  `,
};


const Content = ({ children, fullHeight, fullWidth }) => {
  return (
    <div className={cx(styles.root, {
      [styles.fullHeight]: fullHeight,
    })}>
      <div className={cx(styles.children, { [styles.fullWidth]: fullWidth })}>
        {children}
      </div>
    </div>
  );
};


Content.propTypes = {
  children: PropTypes.node.isRequired,

  /** If true, the content will take all the height available */
  fullHeight: PropTypes.bool,

  /** If true, the content will not be limited to 1200px */
  fullWidth: PropTypes.bool,
};


export default Content;
