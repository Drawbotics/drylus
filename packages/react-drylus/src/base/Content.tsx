import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import React from 'react';

import { Style } from '../types';

const styles = {
  root: css`
    background: ${sv.backgroundColor};
  `,
  fullHeight: css`
    flex: 1;
  `,
  children: css`
    max-width: ${sv.maxWidthLarge};
    margin: auto;
  `,
  fullWidth: css`
    max-width: none;
    height: 100%;
  `,
};

interface ContentProps {
  children: React.ReactNode;

  /** If true, the content will take all the height available */
  fullHeight?: boolean;

  /** If true, the content will not be limited to 1200px */
  fullWidth?: boolean;

  /** Used for style overrides */
  style?: Style;
}

export const Content = ({ children, fullHeight, fullWidth, style }: ContentProps) => {
  return (
    <div
      style={style}
      className={cx(styles.root, {
        [styles.fullHeight]: fullHeight,
      })}>
      <div className={cx(styles.children, { [styles.fullWidth]: fullWidth })}>{children}</div>
    </div>
  );
};
