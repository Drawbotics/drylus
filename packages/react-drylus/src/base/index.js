export ThemeProvider from './ThemeProvider';
export Page from './Page';
export Sizes from './Sizes';
export Categories from './Categories';
