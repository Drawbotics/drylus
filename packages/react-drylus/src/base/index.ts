export * from './DrylusProvider';
export * from './ThemeProvider';
export * from './Page';
export * from './Content';
