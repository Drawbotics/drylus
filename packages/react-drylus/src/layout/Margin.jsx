import React from 'react';
import PropTypes from 'prop-types';
import { css, cx } from 'emotion';
import sv from '@drawbotics/style-vars';
import camelCase from 'lodash/camelCase';

import Sizes from '../base/Sizes';


const styles = {
  root: css`
    margin: ${sv.defaultMargin};
  `,
  resetMargin: css`
    margin: 0;
  `,
  extraSmall: css`
    margin: ${sv.marginExtraSmall};
  `,
  small: css`
    margin: ${sv.marginSmall};
  `,
  large: css`
    margin: ${sv.marginLarge};
  `,
  extraLarge: css`
    margin: ${sv.marginExtraLarge};
  `,
  extraSmallLeft: css`
    margin-left: ${sv.marginExtraSmall};
  `,
  extraSmallRight: css`
    margin-right: ${sv.marginExtraSmall};
  `,
  extraSmallTop: css`
    margin-top: ${sv.marginExtraSmall};
  `,
  extraSmallBottom: css`
    margin-bottom: ${sv.marginExtraSmall};
  `,
  smallLeft: css`
    margin-left: ${sv.marginSmall};
  `,
  smallRight: css`
    margin-right: ${sv.marginSmall};
  `,
  smallTop: css`
    margin-top: ${sv.marginSmall};
  `,
  smallBottom: css`
    margin-bottom: ${sv.marginSmall};
  `,
  defaultLeft: css`
    margin-left: ${sv.defaultMargin};
  `,
  defaultRight: css`
    margin-right: ${sv.defaultMargin};
  `,
  defaultTop: css`
    margin-top: ${sv.defaultMargin};
  `,
  defaultBottom: css`
    margin-bottom: ${sv.defaultMargin};
  `,
  largeLeft: css`
    margin-left: ${sv.marginLarge};
  `,
  largeRight: css`
    margin-right: ${sv.marginLarge};
  `,
  largeTop: css`
    margin-top: ${sv.marginLarge};
  `,
  largeBottom: css`
    margin-bottom: ${sv.marginLarge};
  `,
  extraLargeLeft: css`
    margin-left: ${sv.marginExtraLarge};
  `,
  extraLargeRight: css`
    margin-right: ${sv.marginExtraLarge};
  `,
  extraLargeTop: css`
    margin-top: ${sv.marginExtraLarge};
  `,
  extraLargeBottom: css`
    margin-bottom: ${sv.marginExtraLarge};
  `,
};


const Margin = ({
  children,
  size,
  style,
}) => {
  const isUniform = typeof size !== 'object';
  return (
    <div className={cx(styles.root, {
      [styles[camelCase(size)]]: isUniform && size,
      [styles.resetMargin]: ! isUniform,
      [styles[camelCase(`${size.left}_LEFT`)]]: ! isUniform && size.left,
      [styles[camelCase(`${size.right}_RIGHT`)]]: ! isUniform && size.right,
      [styles[camelCase(`${size.top}_TOP`)]]: ! isUniform && size.top,
      [styles[camelCase(`${size.bottom}_BOTTOM`)]]: ! isUniform && size.bottom,
    })} style={style}>
      {children}
    </div>
  );
};


Margin.propTypes = {
  /** Determines the amount of margin given to the component. If a single value, the margin is applied equally to each side */
  size: PropTypes.oneOfType([
    PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
    PropTypes.shape({
      left: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      right: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      bottom: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
      top: PropTypes.oneOf([ Sizes.DEFAULT, Sizes.SMALL, Sizes.EXTRA_SMALL, Sizes.LARGE, Sizes.EXTRA_LARGE ]),
    }),
  ]),

  /** The content of the margin wrapper */
  children: PropTypes.node,

  /** Prop to override any style if necessary, use sparingly */
  style: PropTypes.object,
};


export default Margin;
