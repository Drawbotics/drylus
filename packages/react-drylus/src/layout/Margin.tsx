import sv from '@drawbotics/drylus-style-vars';
import { css, cx } from 'emotion';
import camelCase from 'lodash/camelCase';
import React from 'react';

import { Size } from '../enums';
import { Rectangular, Responsive, Style, Variable } from '../types';
import { useResponsiveProps } from '../utils';

const styles = {
  root: css`
    margin: ${sv.defaultMargin};

    @media ${sv.screenL} {
      margin: ${sv.marginSmall};
    }
  `,
  resetMargin: css`
    margin: 0;

    @media ${sv.screenL} {
      margin: 0;
    }
  `,
  extraSmall: css`
    margin: ${sv.marginExtraSmall};

    @media ${sv.screenL} {
      margin: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  small: css`
    margin: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin: ${sv.marginExtraSmall};
    }
  `,
  large: css`
    margin: ${sv.marginLarge};

    @media ${sv.screenL} {
      margin: ${sv.defaultMargin};
    }
  `,
  extraLarge: css`
    margin: ${sv.marginExtraLarge};

    @media ${sv.screenL} {
      margin: ${sv.marginLarge};
    }
  `,
  huge: css`
    margin: ${sv.marginHuge};

    @media ${sv.screenL} {
      margin: ${sv.marginExtraLarge};
    }
  `,
  extraHuge: css`
    margin: ${sv.marginExtraHuge};

    @media ${sv.screenL} {
      margin: ${sv.marginHuge};
    }
  `,
  massive: css`
    margin: ${sv.marginMassive};

    @media ${sv.screenL} {
      margin: ${sv.marginExtraHuge};
    }
  `,
  extraSmallLeft: css`
    margin-left: ${sv.marginExtraSmall};

    @media ${sv.screenL} {
      margin-left: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  extraSmallRight: css`
    margin-right: ${sv.marginExtraSmall};

    @media ${sv.screenL} {
      margin-right: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  extraSmallTop: css`
    margin-top: ${sv.marginExtraSmall};

    @media ${sv.screenL} {
      margin-top: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  extraSmallBottom: css`
    margin-bottom: ${sv.marginExtraSmall};

    @media ${sv.screenL} {
      margin-bottom: calc(${sv.marginExtraSmall} / 2);
    }
  `,
  smallLeft: css`
    margin-left: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-left: ${sv.marginExtraSmall};
    }
  `,
  smallRight: css`
    margin-right: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-right: ${sv.marginExtraSmall};
    }
  `,
  smallTop: css`
    margin-top: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-top: ${sv.marginExtraSmall};
    }
  `,
  smallBottom: css`
    margin-bottom: ${sv.marginSmall};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginExtraSmall};
    }
  `,
  defaultLeft: css`
    margin-left: ${sv.defaultMargin};

    @media ${sv.screenL} {
      margin-left: ${sv.marginSmall};
    }
  `,
  defaultRight: css`
    margin-right: ${sv.defaultMargin};

    @media ${sv.screenL} {
      margin-right: ${sv.marginSmall};
    }
  `,
  defaultTop: css`
    margin-top: ${sv.defaultMargin};

    @media ${sv.screenL} {
      margin-top: ${sv.marginSmall};
    }
  `,
  defaultBottom: css`
    margin-bottom: ${sv.defaultMargin};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginSmall};
    }
  `,
  largeLeft: css`
    margin-left: ${sv.marginLarge};

    @media ${sv.screenL} {
      margin-left: ${sv.defaultMargin};
    }
  `,
  largeRight: css`
    margin-right: ${sv.marginLarge};

    @media ${sv.screenL} {
      margin-right: ${sv.defaultMargin};
    }
  `,
  largeTop: css`
    margin-top: ${sv.marginLarge};

    @media ${sv.screenL} {
      margin-top: ${sv.defaultMargin};
    }
  `,
  largeBottom: css`
    margin-bottom: ${sv.marginLarge};

    @media ${sv.screenL} {
      margin-bottom: ${sv.defaultMargin};
    }
  `,
  extraLargeLeft: css`
    margin-left: ${sv.marginExtraLarge};

    @media ${sv.screenL} {
      margin-left: ${sv.marginLarge};
    }
  `,
  extraLargeRight: css`
    margin-right: ${sv.marginExtraLarge};

    @media ${sv.screenL} {
      margin-right: ${sv.marginLarge};
    }
  `,
  extraLargeTop: css`
    margin-top: ${sv.marginExtraLarge};

    @media ${sv.screenL} {
      margin-top: ${sv.marginLarge};
    }
  `,
  extraLargeBottom: css`
    margin-bottom: ${sv.marginExtraLarge};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginLarge};
    }
  `,
  hugeLeft: css`
    margin-left: ${sv.marginHuge};

    @media ${sv.screenL} {
      margin-left: ${sv.marginExtraLarge};
    }
  `,
  hugeRight: css`
    margin-right: ${sv.marginHuge};

    @media ${sv.screenL} {
      margin-right: ${sv.marginExtraLarge};
    }
  `,
  hugeTop: css`
    margin-top: ${sv.marginHuge};

    @media ${sv.screenL} {
      margin-top: ${sv.marginExtraLarge};
    }
  `,
  hugeBottom: css`
    margin-bottom: ${sv.marginHuge};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginExtraLarge};
    }
  `,
  extraHugeLeft: css`
    margin-left: ${sv.marginExtraHuge};

    @media ${sv.screenL} {
      margin-left: ${sv.marginHuge};
    }
  `,
  extraHugeRight: css`
    margin-right: ${sv.marginExtraHuge};

    @media ${sv.screenL} {
      margin-right: ${sv.marginHuge};
    }
  `,
  extraHugeTop: css`
    margin-top: ${sv.marginExtraHuge};

    @media ${sv.screenL} {
      margin-top: ${sv.marginHuge};
    }
  `,
  extraHugeBottom: css`
    margin-bottom: ${sv.marginExtraHuge};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginHuge};
    }
  `,
  massiveLeft: css`
    margin-left: ${sv.marginMassive};

    @media ${sv.screenL} {
      margin-left: ${sv.marginExtraHuge};
    }
  `,
  massiveRight: css`
    margin-right: ${sv.marginMassive};

    @media ${sv.screenL} {
      margin-right: ${sv.marginExtraHuge};
    }
  `,
  massiveTop: css`
    margin-top: ${sv.marginMassive};

    @media ${sv.screenL} {
      margin-top: ${sv.marginExtraHuge};
    }
  `,
  massiveBottom: css`
    margin-bottom: ${sv.marginMassive};

    @media ${sv.screenL} {
      margin-bottom: ${sv.marginExtraHuge};
    }
  `,
};

function _computeSize(sizeDescription: Rectangular) {
  let size: Variable = {};

  if (sizeDescription.horizontal) {
    size.left = sizeDescription.horizontal;
    size.right = sizeDescription.horizontal;
  }
  if (sizeDescription.vertical) {
    size.top = sizeDescription.vertical;
    size.bottom = sizeDescription.vertical;
  }

  return size;
}

function _isRectangular(size: any): size is Rectangular {
  return size.vertical || size.horizontal;
}

export interface MarginProps {
  /** Determines the amount of margin given to the component. If a single value, the margin is applied equally to each side */
  size?: Size | Rectangular | Variable;

  /** The content of the margin wrapper */
  children?: React.ReactNode;

  /** Prop to override any style if necessary, use sparingly */
  style?: Style;

  /** Used for style overrides */
  className?: string;

  /** Reponsive prop overrides */
  responsive?: Responsive<this>;
}

export const Margin = ({ responsive, ...rest }: MarginProps) => {
  const { children, size: rawSize, style, className } = useResponsiveProps<MarginProps>(
    rest,
    responsive,
  );

  const isUniform = typeof rawSize !== 'object';

  const size = !isUniform && _isRectangular(rawSize) ? _computeSize(rawSize) : rawSize;
  return (
    <div
      className={cx(
        styles.root,
        {
          [styles[camelCase(size as Size) as keyof typeof styles]]: isUniform && size != null,
          [styles.resetMargin]: !isUniform,
          [styles[camelCase(`${(size as Variable)?.left}_LEFT`) as keyof typeof styles]]:
            !isUniform && (size as Variable)?.left != null,
          [styles[camelCase(`${(size as Variable)?.right}_RIGHT`) as keyof typeof styles]]:
            !isUniform && (size as Variable)?.right != null,
          [styles[camelCase(`${(size as Variable)?.top}_TOP`) as keyof typeof styles]]:
            !isUniform && (size as Variable)?.top != null,
          [styles[camelCase(`${(size as Variable)?.bottom}_BOTTOM`) as keyof typeof styles]]:
            !isUniform && (size as Variable)?.bottom != null,
        },
        className,
      )}
      style={style}>
      {children}
    </div>
  );
};
