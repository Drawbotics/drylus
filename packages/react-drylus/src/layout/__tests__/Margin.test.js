import React from 'react';
import { create } from 'react-test-renderer';

import Margin from '../Margin';
import { Sizes } from '../../base';


describe('Margin', () => {
  describe('matches snapshot when', () => {
    it('has a uniform size', () => {
      const tree = create(
        <Margin size={Sizes.LARGE}>
          Content
        </Margin>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a size only on 1 side', () => {
      const tree = create(
        <Margin size={{ left: Sizes.LARGE }}>
          Content
        </Margin>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has a different size on each side', () => {
      const tree = create(
        <Margin
          size={{
            left: Sizes.LARGE,
            top: Sizes.SMALL,
            bottom: Sizes.DEFAULT,
            right: Sizes.EXTRA_SMALL,
          }}>
          Content
        </Margin>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('has vertical and horizontal sizes', () => {
      const tree = create(
        <Margin
          size={{
            vertical: Sizes.LARGE,
            horizontal: Sizes.SMALL,
          }}>
          Content
        </Margin>
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});