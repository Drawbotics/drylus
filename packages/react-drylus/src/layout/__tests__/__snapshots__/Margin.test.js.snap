// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Margin matches snapshot when has a different size on each side 1`] = `
.css-znzz4z-Margin__root-Margin__resetMargin {
  margin: 24px;
  margin: 0;
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 0;
  }
}

<div
  className="undefined css-znzz4z-Margin__root-Margin__resetMargin"
>
  Content
</div>
`;

exports[`Margin matches snapshot when has a size only on 1 side 1`] = `
.css-znzz4z-Margin__root-Margin__resetMargin {
  margin: 24px;
  margin: 0;
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 0;
  }
}

<div
  className="css-znzz4z-Margin__root-Margin__resetMargin"
>
  Content
</div>
`;

exports[`Margin matches snapshot when has a uniform size 1`] = `
.css-1elarmb-Margin__root {
  margin: 24px;
}

@media only screen and (max-width:768px) {
  .css-1elarmb-Margin__root {
    margin: 16px;
  }
}

<div
  className="css-1elarmb-Margin__root"
>
  Content
</div>
`;

exports[`Margin matches snapshot when has vertical and horizontal sizes 1`] = `
.css-znzz4z-Margin__root-Margin__resetMargin {
  margin: 24px;
  margin: 0;
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 16px;
  }
}

@media only screen and (max-width:768px) {
  .css-znzz4z-Margin__root-Margin__resetMargin {
    margin: 0;
  }
}

<div
  className="undefined css-znzz4z-Margin__root-Margin__resetMargin"
>
  Content
</div>
`;
